source/MaterialXCore/Document.cpp

```diff
@@ -1094,8 +1101,31 @@ void Document::upgradeVersion(bool applyFutureUpdates)
                     node->setName(newNodeName);
                 }
             }
+        }   
+
+        // Convert all parameters to be inputs. If needed set them to be "uniform".
+        for (ElementPtr e : traverseTree())
+        {
+            InterfaceElementPtr elem = e->asA<InterfaceElement>();
+            if (!elem)
+            {
+                continue;
+            }
+            vector<ElementPtr> children = elem->getChildren();
+            for (ElementPtr child : children)
+            {
+                if (child->getCategory() == "parameter")
+                {
+                    InputPtr newInput = updateChildSubclass<Input>(elem, child);
+                    // TODO: Only make some of these into uniforms 
+                    // instead of all of them
+                    newInput->setIsUniform(true);
```

Comment:
Does this mean that several of the params you changed to inputs should also set uniform equal to true?

Reply:
I suppose it's really the nodedef that dictates if an input should be uniform or not. Since we are about to change many parameters into non-uniform inputs I think we need to look at the nodedef for every node being upgraded here.

This might be tricky though since in finding the nodedef we use signature matching to find it, and the signature is obviously different (that's what we are trying to fix in the first place). Catch-22 :)

But maybe we can first blindly upgrade all parameters into inputs, as done here. Then find the nodedef, and do a second pass to set all uniform flags correct.

Reply:
Yes. That is the plan. This is the first pass to make it 100% equivalent to what was in 1.37 for this PR. Then I'll be looking at the spec changes from Doug to see if there are patterns to use and if not to find a way to manage per-case outliers.